src/features/adr/components/PatientLookup.tsx

```diff
@@ -0,0 +1,182 @@
+import enLocale from "date-fns/locale/en-NZ";
+import { PatientList } from "../components/PatientList";
+
+import { stylesFactory } from "../../../shared/utils";
+import { useTranslations } from "../../../shared/hooks";
+import { usePatientLookup } from "../../patients/hooks";
+import { isValid, parse } from "date-fns";
+import { Patient, SearchParameters } from "../../patients/types";
+
+const useStyles = stylesFactory({
+  fieldContainer: { paddingBottom: 10 },
+  loadingIndicator: { textAlign: "center", padding: 15 },
+});
+
+export interface PatientLookupProps {
+  patient: Patient;
+  setCanContinue: (canContinue: boolean) => void;
+  setPatient: (patient: Patient) => void;
+}
+
+const initialSearchParams: SearchParameters = {
+  firstName: "",
+  lastName: "",
+  dateOfBirth: null,
+};
+
+export const PatientLookup: FC<PatientLookupProps> = ({
+  patient,
+  setCanContinue,
+  setPatient,
+}) => {
+  const classes = useStyles();
+  const [searchParams, setSearchParams] = useState(initialSearchParams);
+  const { messages } = useTranslations();
+
+  const onFirstNameChange = (event: ChangeEvent<HTMLInputElement>) => {
+    setSearchParams({ ...searchParams, firstName: event.target.value });
+  };
+  const onLastNameChange = (event: ChangeEvent<HTMLInputElement>) => {
+    setSearchParams({ ...searchParams, lastName: event.target.value });
+  };
+
+  const onDateChange = (
+    dateValue?: Date | null | undefined,
+    stringValue?: string
+  ) => {
+    const dateOfBirth =
+      dateValue || parse(stringValue || "", "dd/MM/yyyy", new Date());
+    setSearchParams({ ...searchParams, dateOfBirth });
+  };
+
+  const {
+    data: patientData,
+    error,
+    noMore,
+    loading,
+    searchedWithNoResults,
+    searchOnline,
+    searchMore,
+  } = usePatientLookup();
+
+  const onSelect = (patient: Patient) => {
+    const { ID, name } = patient;
+    setPatient({ ID, name });
+    setCanContinue(true);
+  };
+
+  const onWaypoint = () => {
+    if (noMore || loading) return;
+    searchMore(searchParams);
+  };
+
+  const hasNoSearchParams = useMemo(() => {
+    const { firstName, lastName, dateOfBirth } = searchParams || {};
+    return !firstName && !lastName && !isValid(dateOfBirth);
+  }, [searchParams]);
+
+  const lookupPatients = useMemo(
+    () => () => {
+      if (loading || hasNoSearchParams) return;
+      searchOnline(searchParams);
+    },
+    [searchOnline, hasNoSearchParams, loading, searchParams]
+  );
+
+  useEffect(() => {
+    const listener = (event: any) => {
+      if (event.code === "Enter" || event.code === "NumpadEnter") {
+        event.preventDefault();
+        lookupPatients();
+      }
+    };
+    document.addEventListener("keydown", listener);
+    return () => {
+      document.removeEventListener("keydown", listener);
+    };
+  }, [lookupPatients]);
+
+  return (
+    <Grid container direction="row" justifyContent="space-around">
```

Comment:
218a0d0

now working better. The styling on the `TextFields` aren't great because material overrides margin and padding, so I've wrapped them in `Box`es instead which is more of a material approach, I think.